packages/assets-controllers/src/TokensController.ts

```diff
@@ -693,11 +693,16 @@ export class TokensController extends BaseControllerV1<
   }
 
   _getProvider(networkClientId?: NetworkClientId): Web3Provider {
-    return new Web3Provider(
-      networkClientId
-        ? this.getNetworkClientById(networkClientId).provider
-        : this.config?.provider,
-    );
+    const provider = networkClientId
+      ? this.getNetworkClientById(networkClientId).provider
+      : this.config?.provider;
```

Comment:
nit:
`internalConfig` is always defined in BaseControllerV1
```suggestion
      : this.config.provider;
```